node-gui/src/main_window/main_widget/tabs/wallet.rs

```diff
@@ -0,0 +1,555 @@
+                    .on_press(WalletMessage::RemovePassword)
+            ],
+            EncryptionState::Disabled => row![iced::widget::button(Text::new("Encrypt wallet"))
+                .on_press(WalletMessage::SetPassword),],
+        }
+        .align_items(Alignment::Center)
+        .spacing(10);
+
+    row![
+        balance,
+        Row::new().width(Length::Fill),
+        password,
+        button(Text::new("Close"))
+            .style(iced::theme::Button::Destructive)
+            .on_press(WalletMessage::Close),
+    ]
+    .width(Length::Fill)
+    .height(Length::Fill)
+    .spacing(10)
+    .padding(10)
+    .align_items(Alignment::Center)
+    .into()
+}
+
+#[derive(Clone, PartialEq, Eq)]
+struct AccountPickItem {
+    account_id: AccountId,
+    name: String,
+}
+
+impl std::fmt::Display for AccountPickItem {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        std::fmt::Display::fmt(&self.name, f)
+    }
+}
+
+fn view_left_panel(
+    node_state: &NodeState,
+    wallet_info: &WalletInfo,
+    selected_account: AccountId,
+    selected_panel: SelectedPanel,
+) -> Element<'static, WalletMessage> {
+    let file_name = wallet_info.path.file_name().map_or_else(
+        || "<Unknown>".to_owned(),
+        |file_name| file_name.to_string_lossy().to_string(),
+    );
+
+    let account_items = wallet_info
+        .accounts
+        .iter()
+        .map(|(account_id, account)| AccountPickItem {
+            name: account
+                .name
+                .clone()
+                .unwrap_or_else(|| format!("Account {}", account_id.account_index())),
+            account_id: *account_id,
+        })
+        .collect::<Vec<_>>();
+
+    let selected_account = account_items
+        .iter()
+        .find(|account_item| account_item.account_id == selected_account)
+        .cloned();
+    let pick_list = pick_list(account_items, selected_account, |item| {
+        WalletMessage::SelectAccount(item.account_id)
+    });
+
+    let panel_button = |label, panel, selected_panel| {
+        let label = text(label).size(16);
+
+        let button = button(label)
+            .style(if panel == selected_panel {
+                iced::theme::Button::Primary
+            } else {
+                iced::theme::Button::Text
+            })
+            .width(Length::Fill);
+
+        button.on_press(WalletMessage::SelectPanel(panel)).padding(8)
+    };
+
+    // `next_height` is used to prevent flickering when a new block is found
+    let show_scan_progress = wallet_info.best_block.1.next_height() < node_state.best_block_height;
+    let scan_progress_widget = if show_scan_progress {
+        // TODO: Fix scan progress when the node is in the initial block download state
+        let scan_progress = wallet_scan_progress(
+            node_state.best_block_height.into_int(),
+            wallet_info.best_block.1.into_int(),
+        );
+        let scan_progress_str = format!("{:.0}%", scan_progress);
```

Comment:
Please also add (1234/5555 blocks) to that text. It should be intuitive for people to understand what this is.